rust/stub/src/uefi_log.rs

```diff
@@ -0,0 +1,139 @@
+//! UEFI-compatible logging backend
+
+#![deny(missing_docs)]
+#![deny(clippy::pedantic)]
+#![deny(clippy::missing_docs_in_private_items)]
+
+use alloc::{boxed::Box, string::String};
+use core::fmt::Write;
+use core::ptr::NonNull;
+use log::{Level, LevelFilter, Metadata, Record};
+use uefi::{
+    prelude::*,
+    proto::{console::text::Color, console::text::Output, loaded_image::LoadedImage},
+    Result,
+};
+
+/// Logger that logs to UEFI stdout
+struct UefiLogger {
+    /// Maximum level to log
+    max_level: LevelFilter,
+
+    /// Writer to write messages to
+    writer: Option<NonNull<Output<'static>>>,
+}
+
+impl log::Log for UefiLogger {
+    /// Returns whether or not this logger is active for a given message
+    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
+        metadata.level() <= self.max_level
+    }
+
+    /// Logs a record
+    fn log(&self, record: &Record<'_>) {
+        if !self.enabled(record.metadata()) {
+            return;
+        }
+        let writer = if let Some(mut ptr) = self.writer {
+            unsafe { ptr.as_mut() }
+        } else {
+            return;
+        };
+
+        let foreground = match record.level() {
+            Level::Error => Color::Red,
+            Level::Warn => Color::Yellow,
+            Level::Info => Color::White,
+            Level::Debug => Color::Blue,
+            Level::Trace => Color::Cyan,
+        };
+        let _ = writer.set_color(foreground, Color::Black);
+        let _ = write!(writer, "{}", record.level());
+        let _ = writer.set_color(Color::White, Color::Black);
+        let _ = write!(writer, " - {}\r\n", record.args());
```

Comment:
This seems to be there to ignore errors, which I think is the right choice here. This could be made more obvious to the reader with a short comment before the calls.